misc/peer-store/src/memory_store.rs

```diff
@@ -226,17 +267,20 @@ impl<T> PeerRecord<T> {
     /// Update the address in the LRU cache, promote it to the front if it exists,
     /// insert it to the front if not.
     /// Returns true when the address is new.
-    pub fn update_address(&mut self, address: &Multiaddr) -> bool {
+    pub fn update_address(&mut self, address: &Multiaddr, permanent: bool) -> bool {
         if self.addresses.get(address).is_some() {
             return false;
         }

```

Comment:
If `permanent` is `true` I think we should update the `permanent` parameter on the address. Otherwise there is no way for the user to change an existing discovered address into a permanent one.